src/components/top-bar-equipment-seach-dialog/use-search-matching-equipments.tsx

```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import { useCallback, useEffect, useRef, useState } from 'react';
+import {
+    EquipmentInfos,
+    EquipmentType,
+    Identifiable,
+    getEquipmentsInfosForSearchBar,
+    useSnackMessage,
+    useDebounce,
+    Equipment,
+} from '@gridsuite/commons-ui';
+import { SEARCH_FETCH_TIMEOUT_MILLIS } from '../../utils/UIconstants';
+import { useNameOrId } from '../utils/equipmentInfosHandler';
+import { searchEquipmentsInfos } from '../../services/study';
+import { UUID } from 'crypto';
+
+interface UseSearchMatchingEquipmentsProps {
+    studyUuid: UUID;
+    nodeUuid: UUID;
+    inUpstreamBuiltParentNode?: boolean;
+    equipmentType?: EquipmentType;
+    makeItems?: (
+        equipmentsInfos: Equipment[],
+        getNameOrId: (e: Identifiable) => string
+    ) => EquipmentInfos[];
+}
+
+export const useSearchMatchingEquipments = (
+    props: UseSearchMatchingEquipmentsProps
+) => {
+    const {
+        studyUuid,
+        nodeUuid,
+        inUpstreamBuiltParentNode,
+        equipmentType,
+        makeItems = getEquipmentsInfosForSearchBar,
```

Comment:
well it's never defined to another value than this default `getEquipmentsInfosForSearchBar`, maybe we could simplify removing this prop and calling `getEquipmentsInfosForSearchBar` directly ?

Reply:
I agree, fixed